component/vdr/cheqd/vdr.go

```diff
@@ -0,0 +1,245 @@
+	}
+
+	doc, err := did.ParseDocument(body)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:cheqd did --> error parsing did doc --> %w", err)
+	}
+
+	return &did.DocResolution{DIDDocument: doc}, nil
+}
+
+// RawDoc type.
+type rawDoc struct {
+	Context              interface{}              `json:"@context,omitempty"`
+	ID                   string                   `json:"id,omitempty"`
+	VerificationMethod   []map[string]interface{} `json:"verificationMethod,omitempty"`
+	PublicKey            []map[string]interface{} `json:"publicKey,omitempty"`
+	Service              []map[string]interface{} `json:"service,omitempty"`
+	Authentication       []interface{}            `json:"authentication,omitempty"`
+	AssertionMethod      []interface{}            `json:"assertionMethod,omitempty"`
+	CapabilityDelegation []interface{}            `json:"capabilityDelegation,omitempty"`
+	CapabilityInvocation []interface{}            `json:"capabilityInvocation,omitempty"`
+	KeyAgreement         []interface{}            `json:"keyAgreement,omitempty"`
+	Created              *time.Time               `json:"created,omitempty"`
+	Updated              *time.Time               `json:"updated,omitempty"`
+	Proof                []interface{}            `json:"proof,omitempty"`
+}
+
+// RawDocCheqd type.
+type RawDocCheqd struct {
+	DidDocument           DidDocument           `json:"didDocument"`
+	DidDocumentMetadata   DidDocumentMetadata   `json:"didDocumentMetadata"`
+	DidResolutionMetadata DidResolutionMetadata `json:"didResolutionMetadata"`
+}
+
+// DidDocument type.
+type DidDocument struct {
+	Context            []string             `json:"@context"`
+	Authentication     []string             `json:"authentication"`
+	ID                 string               `json:"id"`
+	Service            []Service            `json:"service"`
+	VerificationMethod []VerificationMethod `json:"verificationMethod"`
+}
+
+// Service type.
+type Service struct {
+	ID              string `json:"id"`
+	ServiceEndpoint string `json:"serviceEndpoint"`
+	Type            string `json:"type"`
+}
+
+// VerificationMethod type.
+type VerificationMethod struct {
+	Controller         string `json:"controller"`
+	ID                 string `json:"id"`
+	PublicKeyMultibase string `json:"publicKeyMultibase"`
+	Type               string `json:"type"`
+}
+
+// DidDocumentMetadata type.
+type DidDocumentMetadata struct {
+	Created   string `json:"created"`
+	VersionID string `json:"versionId"`
+}
+
+// DidResolutionMetadata type.
+type DidResolutionMetadata struct {
+	ContentType string `json:"contentType"`
+	Retrieved   string `json:"retrieved"`
+	DID         DID    `json:"did"`
+}
+
+// DID type.
+type DID struct {
+	DIDString        string `json:"didString"`
+	MethodSpecificID string `json:"methodSpecificId"`
+	Method           string `json:"method"`
+}
+
+func convertTypeRawDocCheqdToRawDoc(rawDocCheqd *RawDocCheqd) *rawDoc {
```

Comment:
Why can't we just Unmarshal a doc returned by cheqd to a rawDoc expected by Aries-framework-go? Why do we need that intermediate structs?

Reply:
We can't do that, because what we get where we send their fields are different

Reply:
I don't see any differences in af-go rawDoc and "didDocument" content of the cheqd, and there should not be any difference as it's W3C standard (it there is a difference, then some party violates the standard).
So, do we really need to declare DidDocument, Service, VerificationMethod, etc.? Why don't just do
```
type RawDocCheqd struct {
 	DidDocument           rawDoc           `json:"didDocument"` // af-go defined struct
 	DidDocumentMetadata   DidDocumentMetadata   `json:"didDocumentMetadata"`
 	DidResolutionMetadata DidResolutionMetadata `json:"didResolutionMetadata"`
 }
```